src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -375,7 +460,9 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 				let { key: pattern, value: ids } = entry;
 				if (match(pattern, uri.fsPath)) {
 					for (let id of ids) {
-						this._fileBasedRecommendations[id] = now;
+						if (this.isExtensionAllowedToBeRecommended(id)) {
```

Comment:
we dont need to do this if `_availableRecommendations` gets updated in `refreshAllIgnoredRecommendations`